src/main/java/co/omise/models/Refund.java

```diff
@@ -56,20 +61,87 @@ public void setMetadata(Map<String, Object> metadata) {
         this.metadata = metadata;
     }
 
-    public static class Create extends Params {
+    /**
+     * The {@link RequestBuilder} class for creating a Refund.
+     */
+    public static class CreateRequestBuilder extends RequestBuilder<Refund> {
+        private String chargeId;
         @JsonProperty
         private long amount;
         @JsonProperty
         private Map<String, Object> metadata;
 
-        public Create amount(long amount) {
+        public CreateRequestBuilder(String chargeId) {
+            this.chargeId = chargeId;
+        }
+
+        @Override
+        protected HttpUrl path() {
+            return buildUrl(Endpoint.API, "charges", chargeId, "refunds");
+        }
+
+        @Override
+        protected String method() {
+            return POST;
+        }
+
+        @Override
+        protected RequestBody payload() throws IOException {
+            return serialize();
+        }
+
+        public CreateRequestBuilder amount(long amount) {
             this.amount = amount;
             return this;
         }
 
-        public Create metadata(Map<String, Object> metadata) {
+        public CreateRequestBuilder metadata(Map<String, Object> metadata) {
             this.metadata = metadata;
```

Comment:
Let's have both metadata(Map<> map) and metadata(String string, Object object) methods for this. Have a look at the way I've done it on the `Token` PR.

Reply:
🤔 @pitiphong-p @chakrit Do we need both `metadata(Map<> map)` and `metadata(String string, Object object) `?

Reply:
Building maps is not that trivial in java. Requires at least 3 lines where the default case usually is just one key, one value ("invoice_id", "123") merchant ease of use wins here.